### ISSUE TITLE
'v8::String::Value::Value': no overloaded function takes 2 arguments

### ISSUE THREAD

**Issue by 4mfie** [atom/node-oniguruma#96]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: <https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode>
    * Followed all applicable steps in the debugging guide: <https://flight-manual.atom.io/hacking-atom/sections/debugging/>
    * Checked the FAQs on the message board for common solutions: <https://discuss.atom.io/c/faq>
    * Checked that your issue isn't already filed: <https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom>
    * Checked that there is not already an Atom package that provides the described functionality: <https://atom.io/packages>

### Description

When installing `node-oniguruma` v7.2.0 on Windows 10, the following error occurs:

```
..\src\onig-string.cc(37): error C2661: 'v8::String::Value::Value': no overloaded function takes 2 arguments [C:\Users\
REDACTED\source\repos\node-oniguruma\build\onig_scanner.vcxproj]
```

Actual username redacted.

### Steps to Reproduce

1. Install latest Node and NVM for Windows
2. Use NVM to download Node 7.10.1 and then switch to using that version
3. Install Visual Studio Community 2017 w/ the Desktop development with C++ workload and the VC++ 2015.3 v14.00 (v140) toolset for desktop component
4. Set the `VCTargetsPath` environment variable to `C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets`
5. Open Windows PowerShell as administrator
6. `git clone https://github.com/atom/node-oniguruma oniguruma`
7. `cd oniguruma`
8. `git checkout tags/v7.2.0`
9. `npm i`

**Expected behavior:**

Install without receiving any errors.

**Actual behavior:**

Receive the following error:
```
..\src\onig-string.cc(37): error C2661: 'v8::String::Value::Value': no overloaded function takes 2 arguments [C:\Users\
REDACTED\source\repos\node-oniguruma\build\onig_scanner.vcxproj]
```

Actual username redacted.

**Reproduces how often:**

Every single time.

### Versions

`node --version`: `v7.10.1`
`npm --version`: `4.2.0`
`nvm --version`: `Running version 1.1.7.`

OS: Windows 10 Education
System type: 64 bit operating system, x64-based processor
Version: 1803
OS Build: 17134.829

### Additional Information

Editing the `src/onig-string.cc` file by commenting out lines 35-38 & 40 and then running `npm i` seems to resolve the issue.

As this error does not occur with the previous version of `oniguruma` (v7.2.0), my temporary workaround for [ampproject/ampstart](https://github.com/ampproject/ampstart) is to downgrade and lock the package that depends on `oniguruma` v7.2.0 to its previous version (prior to the version bump). My proposal can be seen in the following pull request: [ampproject/ampstart#878](https://github.com/ampproject/ampstart/pull/878), more specifically [this commit](https://github.com/ampproject/ampstart/pull/878/commits/55646959de43cdafbc55b99118a5f9761bad4b19).

**Comment by rsese** [comment]
> Editing the `src/onig-string.cc` file by commenting out lines 35-38 & 40 and then running `npm i` seems to resolve the issue.

Hmm, it looks like these lines were added in https://github.com/atom/node-oniguruma/pull/93 @rafeca?  Do you think this is a problem with that PR?

**Comment by rafeca** [comment]
Thanks for letting us know @amphyploid!

While adding support for Node.js v12 in #93 we dropped the support for Node.js v6, so now the minimum supported version is v8.

This was a breaking change and should have been published as v8.0.0, apologies for that...

Can you update your system to use Node v8 or greater? node v6/v7 [are no longer maintained](https://nodejs.org/en/about/releases/) so libraries are going to progressively drop support for it.

**Comment by 4mfie** [comment]
Would be best to ask @camelburrito that question. He's in charge. I'll go ahead and close this issue for now.
